cdc/entry/mounter.go

```diff
@@ -442,25 +331,32 @@ func datum2Column(tableInfo *model.TableInfo, datums map[int64]types.Datum, fill
 }
 
 func (m *mounterImpl) mountRowKVEntry(tableInfo *model.TableInfo, row *rowKVEntry, dataSize int64) (*model.RowChangedEvent, error) {
-	// if m.enableOldValue == true, go into this function
-	// if m.enableNewValue == false and row.Delete == false, go into this function
-	// if m.enableNewValue == false and row.Delete == true and use explict row id, go into this function
-	// only if m.enableNewValue == false and row.Delete == true and use implicit row id(_tidb_rowid), skip this function
-	useImplicitTiDBRowID := !tableInfo.PKIsHandle && !tableInfo.IsCommonHandle
-	if !m.enableOldValue && row.Delete && useImplicitTiDBRowID {
-		return nil, nil
-	}
-
 	var err error
 	// Decode previous columns.
 	var preCols []*model.Column
+	// Since we now always use old value internally,
+	// we need to control the output(sink will use the PreColumns field to determine whether to output old value).
+	// Normally old value is output when only enableOldValue is on,
+	// but for the Delete event, when the old value feature is off,
+	// the HandleKey column needs to be included as well. So we need to do the following filtering.
 	if row.PreRowExist {
 		// FIXME(leoppro): using pre table info to mounter pre column datum
 		// the pre column and current column in one event may using different table info
```

Comment:
Look like it's a bug, could you report an issue to track this FIXME? @hi-rustin

Reply:
I will file an issue for it.

Reply:
Created. https://github.com/pingcap/ticdc/issues/2302